presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -192,7 +193,8 @@ public boolean needsInput()
                 && lookupSourceProviderFuture.isDone()
                 && spillInProgress.isDone()
                 && probe == null
-                && outputPage == null;
+                && outputPage == null
+                && !isLookupSourceEmpty();
```

Comment:
Check this right after `lookupSourceProviderFuture.isDone()`... or better yet, don't check it here at all, check in `tryFetchLookupSourceProvider`